### PR TITLE
Added move and rename for files

### PR DESCRIPTION
**Pull request by praneethpr**

### CODE REVIEW

**Review by fightingtexasaggie — APPROVED** [review]
This looks pretty solid. Both the rename file in place and move worked, and the resulting submission contained the expected directory structure on disk. This is pretty impressive.

File: src/Controller/Api/FileController.php [hunk 1]
```diff
@@ -51,7 +53,48 @@ public function deleteFile(File $file, MessageBusInterface $messageBus, EntityMa
...
+     * @return Response A response object with an empty body and a "no content" status code.
+     */
+    public function updateFileName(File $file, EntityManagerInterface $entityManager, Request $request, Datastore $datastore) : Response
+    {
+        $newFileName = $request->get('destinationDir');
+        $fileset = $file->getFileset();
+        if (!$fileset->doesFileExist($newFileName)) {
+            // Rename file on disk if it is already processed
+            if ($file->getStatus() === File::FILE_DONE) {
+                $datastore->renameFile($file->getFilePathName(), $newFileName);
```
Comment (praneethpr) [inline src/Controller/Api/FileController.php:87]: First parameter needs to be physicalFilePath, and renameFile needs to make a newFileName based on Datastore standards

File: src/Controller/Api/FileController.php [hunk 2]
```diff
@@ -51,7 +53,48 @@ public function deleteFile(File $file, MessageBusInterface $messageBus, EntityMa
...
+            }
+            $file->setFilePathName($newFileName);
+            $entityManager->flush();
+        } else {
+            throw new BadRequestHttpException('File with same name and folder already exists');
+        }
+
+        return new Response(
+            null,
+            Response::HTTP_OK,
```
Comment (praneethpr) [inline src/Controller/Api/FileController.php:97]: Return HTTP_NO CONTENT

File: src/Util/Datastore.php [hunk 3]
```diff
@@ -111,6 +108,22 @@ public function deleteFile(string $filePath): bool
      */
     public function renameFile(string $oldFilePath, string $newFilePath): bool
     {
-        return $this->datastoreFlysystem->rename($oldFilePath, $newFilePath);
+        return $this->datastoreFlysystem->rename($oldFilePath, $this->makeFileName($newFilePath));
```
Comment (fightingtexasaggie) [inline src/Util/Datastore.php:111]: We also need to inform the database of the filename change.

**Pull request merged**